R/information-matrices.R

```diff
@@ -265,6 +326,35 @@ Fisher_info <- function(mod, type = "expected") {
   }
 
   rownames(info) <- colnames(info) <- theta_names
+
+  # info mat for seperate variances
+  if (!is.null(mod$call$weights) & separate_variances) {
+    varFunc <- sub("\\(.*", "", mod$call$weights)[1]
+    if (varFunc == "varIdent") {
+      theta_reparam <- extract_varcomp(mod, separate_variances = TRUE)
+      theta_reparam_names <- vapply(strsplit(names(unlist(theta_reparam)), split = "[.]"),
+                            function(x) paste(unique(x), collapse = "."), character(1L))
+      r12 <- length(unlist(theta[c(1,2)]))
+      r34 <- length(unlist(theta[c(3,4)]))
+      r <- length(unlist(theta))
+      Jac_1 <- diag(1, r12)
+      Jac_2 <- matrix(0, nrow = r12, ncol = r34)
+      Jac_3 <- t(Jac_2)
+      Jac_41 <- rep(0, length(unlist(theta[3])))
+      Jac_42 <- 1
+      Jac_43 <- diag(as.numeric(theta[4])*2*(unlist(theta[3])), length(unlist(theta[3])))
+      Jac_44 <- as.numeric(unlist(theta[3])^2)
+      Jac_4 <- matrix(rbind(c(Jac_41,Jac_42), cbind(Jac_43, Jac_44)), nrow = r34)
```

Comment:
Those might be simplified. And I used $I^E_{reparam} = (J^T)^{-1} * I^E * J^{-1}$. Is this right or is there a better way to obtain the Fisher information matrix for the re-parameterization?

Reply:
I’ll take a look tomorrow in more detail. But you could maybe use the fact that the inverse of a derivative is equal to the derivative of the inverse. So instead of taking the Jacobean of a function theta = f(lambda), take the Jacobean of lambda = f^-1(theta).